File: CourseProjects/CourseProjects/VisualStudioProjects/ASP/AspWebFormsExamplesAndExercises/aspx_vb_exercises/App_Code/CourseExercise.vb
Imports Microsoft.VisualBasic

Public Class CourseExercise
    Private ReadOnly _title As String
    Private ReadOnly _description As String
    Private ReadOnly _exerciseUrl As String
    Private ReadOnly _solutionUrl As String

    Public Sub New(ByVal title As String, ByVal description As String, ByVal exerciseUrl As String, ByVal solutionUrl As String)
        _title = title
        _description = description
        _exerciseUrl = exerciseUrl
        _solutionUrl = solutionUrl
    End Sub
    Public ReadOnly Property Title() As String
        Get
            Return _title
        End Get
    End Property

    Public ReadOnly Property Description() As String
        Get
            Return _description
        End Get
    End Property
	    Public ReadOnly Property ExerciseUrl As String
        Get
            Return _exerciseUrl
        End Get
    End Property
	    Public ReadOnly Property SolutionUrl As String
        Get
            Return _solutionUrl
        End Get
    End Property
End Class

File: CourseProjects/CourseProjects/VisualStudioProjects/ASP/AspWebFormsExamplesAndExercises/aspx_cs_exercises/App_Code/AspxExercises/CourseExercise.cs
using System;

namespace AspxExercises {
	public class CourseExercise {
        private readonly string title;
        private readonly string description;
		private readonly string exerciseUrl;
        private readonly string solutionUrl;

	    public CourseExercise(string title, string description, string exerciseUrl, string solutionUrl) {
	        this.title = title;
	        this.description = description;
	        this.exerciseUrl = exerciseUrl;
	        this.solutionUrl = solutionUrl;
	    }
	    public string Title {
	        get { return title; }
	    }
	    public string Description {
	        get { return description; }
	    }
	    public string ExerciseUrl {
	        get { return exerciseUrl; }
	    }
	    public string SolutionUrl {
	        get { return solutionUrl; }
	    }
	}
}